### PR TITLE
[ADD][11.0] Adding module project_workflow

### PR DESCRIPTION
**Pull request by pnajman-modoolar**
Hi,

Module ``project_workflow`` provides functionality to create fully configurable workflow around ``project.task`` by wrapping project.task.type (i.e. task stage) with workflow states.

This module depends on following custom modules:

- [project_portal](https://github.com/OCA/project/pull/364)
- [web_diagram_position](https://github.com/OCA/web/pull/879)
- [web_ir_actions_act_multi](https://github.com/OCA/web/pull/877)
- [web_ir_actions_act_view_reload](https://github.com/OCA/web/pull/878)

Kind Regards,
Petar

### CODE REVIEW

**Comment by pedrobaeza** [comment]
web_diagram_position can be optional AFAIK.

Name can be `project_automatic_workflow` following naming of the module `sale_automatic_workflow` that makes the same thing, but for sales.

**Comment by CasVissers-360ERP** [comment]
When exporting:

```
Traceback (most recent call last):
  File "/home/odoo/src/odoo/odoo/http.py", line 650, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/home/odoo/src/odoo/odoo/http.py", line 310, in _handle_exception
    raise pycompat.reraise(type(exception), exception, sys.exc_info()[2])
  File "/home/odoo/src/odoo/odoo/tools/pycompat.py", line 87, in reraise
    raise value
  File "/home/odoo/src/odoo/odoo/http.py", line 692, in dispatch
    result = self._call_function(**self.params)
  File "/home/odoo/src/odoo/odoo/http.py", line 342, in _call_function
    return checked_call(self.db, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/service/model.py", line 97, in wrapper
    return f(dbname, *args, **kwargs)
  File "/home/odoo/src/odoo/odoo/http.py", line 335, in checked_call
    result = self.endpoint(*a, **kw)
  File "/home/odoo/src/odoo/odoo/http.py", line 936, in __call__
    return self.method(*args, **kw)
  File "/home/odoo/src/odoo/odoo/http.py", line 515, in response_wrap
    response = f(*args, **kw)
  File "/home/odoo/src/odoo/addons/web/controllers/main.py", line 935, in call_button
    action = self._call_kw(model, method, args, {})
  File "/home/odoo/src/odoo/addons/web/controllers/main.py", line 923, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "/home/odoo/src/odoo/odoo/api.py", line 689, in call_kw
    return call_kw_multi(method, model, args, kwargs)
  File "/home/odoo/src/odoo/odoo/api.py", line 680, in call_kw_multi
    result = method(recs, *args, **kwargs)
  File "/home/odoo/src/user/project_workflow/models/project_workflow.py", line 288, in export_workflow
    return wizard.button_export()
  File "/home/odoo/src/user/project_workflow/wizard/workflow_export_wizard.py", line 43, in button_export
    exporter.wkf_write(self.workflow_id, stream, "utf-8")
  File "/home/odoo/src/user/project_workflow/models/project_workflow_xml.py", line 271, in wkf_write
    str = self.to_string(workflow, encoding)
  File "/home/odoo/src/user/project_workflow/models/project_workflow_xml.py", line 285, in to_string
    self._build_xml(workflow, element_tree=True),
  File "/home/odoo/src/user/project_workflow/models/project_workflow_xml.py", line 298, in _build_xml
    root = self.create_workflow_element(workflow)
  File "/home/odoo/src/user/project_workflow/models/project_workflow_xml.py", line 317, in create_workflow_element
    return etree.Element('project-workflow', attributes)
  File "src/lxml/etree.pyx", line 2998, in lxml.etree.Element
  File "src/lxml/apihelpers.pxi", line 131, in lxml.etree._makeElement
  File "src/lxml/apihelpers.pxi", line 119, in lxml.etree._makeElement
  File "src/lxml/apihelpers.pxi", line 300, in lxml.etree._initNodeAttributes
  File "src/lxml/apihelpers.pxi", line 311, in lxml.etree._addAttributeToNode
  File "src/lxml/apihelpers.pxi", line 1437, in lxml.etree._utf8
TypeError: Argument must be bytes or unicode, got 'bool'`
```

**Comment by SalahAdDin** [comment]
We certainly need this module!!!

**Comment by pnajman-modoolar** [comment]
@CasVissers, thanks, I'll fix it asap.

File: project_workflow/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# License LGPLv3.0 or later (https://www.gnu.org/licenses/lgpl-3.0.en.html).
+
+{
+    "name": "Project Workflow",
+    "summary": "This module provides workflow for project tasks",
+    "category": "Project",
+    "version": "11.0.1.0.0",
+    "license": "LGPL-3",
+    "author": "Modoolar, Odoo Community Association (OCA)",
+    "website": "https://www.modoolar.com/",
```
Comment (nikul-serpentcs) [inline project_workflow/__manifest__.py:11]: @pnajman-modoolar IMHO Change URL as per [this](https://github.com/OCA/maintainer-tools/blob/master/template/module/__manifest__.py#L10)

**Pull request closed without merging**